Monaco drops panic: runtime error: invalid memory address or nil pointer dereference message

**Describe the bug**
When we run our pipeline with monaco the following occurs. We don't see this with the first run. The only difference we can see is that this MetricId has an '%' in the name  **"metricId": "builtin:tech.dotnet.perfmon.%TimeInGC"**

Below the error. 

**How to reproduce**
Steps to reproduce the behavior:
See the attached tar.gz for the config that is applied.

**Expected behavior**
We expect monaco that just validate, update if needed and continue normally.



**Log output**
If applicable, add log snippet to help explain your problem.

```
2021-03-09 09:24:17 DEBUG 		Applying config `High GC time` using saas_customertech_dta/anomaly-detection-metrics/HighGCtime.json
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x7851c3]

goroutine 1 [running]:
github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest.requestWithBody(0x880005, 0x3, 0xc0001e6a20, 0x8e, 0x9045a0, 0xc0001d4810, 0xc000028011, 0x60, 0x3)
	/source/pkg/rest/request.go:70 +0xa3
github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest.put(0xc000063c80, 0xc0001e6a20, 0x8e, 0xc000549b80, 0x29a, 0xc000028011, 0x60, 0x0, 0x0, 0x0, ...)
	/source/pkg/rest/request.go:60 +0x11c
github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest.upsertDynatraceObject(0xc000063c80, 0xc0003953b0, 0x4f, 0xc0000263c0, 0xc, 0x90d6c0, 0xc000095ec0, 0xc000549b80, 0x29a, 0xc000028011, ...)
	/source/pkg/rest/config_upload.go:54 +0x2c5
github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest.(*dynatraceClientImpl).UpsertByName(0xc000063c50, 0x90d6c0, 0xc000095ec0, 0xc0000263c0, 0xc, 0xc0002c5e60, 0x0, 0x0, 0x0, 0x0, ...)
	/source/pkg/rest/client.go:173 +0x1f4
github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/deploy.uploadConfig(0x90d720, 0xc000063c50, 0x90fe40, 0xc0004b4280, 0xc000063cb0, 0x90c0a0, 0xc000080550, 0x0, 0x0, 0x0, ...)
	/source/pkg/deploy/deploy.go:245 +0x2a5
github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/deploy.execute(0x90c0a0, 0xc000080550, 0xc00002c8e0, 0x1, 0x1, 0x0, 0x87fd5c, 0x1, 0x0, 0x1, ...)
	/source/pkg/deploy/deploy.go:152 +0xee5
github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/deploy.Deploy(0x87fd5c, 0x1, 0x907640, 0xb84308, 0x7ffc42abeed1, 0x1a, 0x0, 0x0, 0x7ffc42abeeb8, 0x15, ...)
	/source/pkg/deploy/deploy.go:60 +0x645
main.getDeployCommand.func2(0xc0000c6bc0, 0x0, 0x0)
	/source/cmd/monaco/main.go:279 +0x2c9
github.com/urfave/cli/v2.(*Command).Run(0xc0000bc360, 0xc0000c6800, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x4ed
github.com/urfave/cli/v2.(*App).RunContext(0xc0000991e0, 0x90bee0, 0xc0000ac030, 0xc000080050, 0x5, 0x5, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x81f
github.com/urfave/cli/v2.(*App).Run(...)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.RunImpl(0xc000080050, 0x5, 0x5, 0x907640, 0xb84308, 0xc000082058)
	/source/cmd/monaco/main.go:49 +0xb9
main.Run(...)
	/source/cmd/monaco/main.go:37
main.main()
	/source/cmd/monaco/main.go:32 +0x5d

```

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 20.04
 - Monaco: 1.5.0

**Additional context**
[example_config.tar.gz](https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/files/6107712/example_config.tar.gz)

--- a/pkg/rest/config_upload.go
+++ b/pkg/rest/config_upload.go
@@ -20,6 +20,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"net/url"
 	"strings"
 	"time"
 
@@ -155,9 +156,9 @@ func upsertDynatraceObject(client *http.Client, fullUrl string, objectName strin
 
 func joinUrl(urlBase string, path string) string {
 	if strings.HasSuffix(urlBase, "/") {
-		return urlBase + path
+		return urlBase + url.PathEscape(path)
 	}
-	return urlBase + "/" + path
+	return urlBase + "/" + url.PathEscape(path)
 }
 
 func isLocationHeaderAvailable(resp Response) (headerAvailable bool, headerArray []string) {

